src/CommandParser.ts

```diff
@@ -0,0 +1,120 @@
+import { IQuoteCommand, QuoteCommand } from "./commands/Quote";
+
+/* 
+  Enum to keep track the current commands that are implemented and the string needed to execute that command.
+  Must be all lowercase to match.
+*/
+export enum CommandType {
+  Impersonate = "impersonate",
+  Generate = "random",
+  Quote = "quote"
+}
+
+/*
+  Implementation to handle converting a Message into a Command.
+*/
+export class CommandParser {
+  private static PREFIX: string = ":"
+  /*
+    Function to convert a Message to a Command.
+    First check if there is a valid command prefix at the start of a mesage (such as "/").
+    Then get the type of the command and convert the message text into a Command.
+    If successful, populate the discordMessage variable and return the command.
+  */
+  public static parseCommand(message: Message): ICommand | null {
+    // entire message (with prefix, command, and args)
+    const messageText = message.content
+    if(this.validPrefix(messageText)){
+      const commandType = this.getCommandType(messageText)?.toLowerCase()
+      if(commandType != null){
+        const command = this.buildCommandFromType(commandType, messageText)
+        if(command){
+          command.discordMessage = message
+          return command;
+        }
+      }
+    }
+    return null;
+  }
+
+  /*
+    Convert the raw text a user sends into its corresponding Command using the given command type.
+  */
+  public static buildCommandFromType(commandType: string, messageText: string): ICommand | null{
+    const args = messageText.split(" ").slice(1)
+    switch(commandType){
+      case CommandType.Impersonate:
+        return this.parseImpersonateCommand(args)
+      case CommandType.Generate:
+        return this.parseGenerateCommand(args)
+      case CommandType.Quote:
+          return this.parseQuoteCommand(args)
+      // case CommandType.Quote:
+      //   return this.parseQuoteCommand(args)
+      // case CommandType.FutureCommandType:
+      //   return this.<function>
+    }
+    return null
+  }
+
+  /*
+    Specific function to parse Impersonate Command data.
+  */
+  public static parseImpersonateCommand(args: string[]): IImpersonateCommand{
+    const impersonateMessage = args.join(" ")
+    const command = new ImpersonateCommand("863639521564295221", impersonateMessage)
+    return command
+  }
+
+  /*
+    Specific function to parse Generate Command data.
+  */
+  public static parseGenerateCommand(args: string[]): IGenerationCommand{
+    const generateMessage = args[0]
+    const command = new GenerateCommand(generateMessage)
+    return command
+  }
+
+  public static parseQuoteCommand(args: string[]): IQuoteCommand{
+    const generateMessage = args[0]
+    var stringValue = ""
+    for( let i = 1; i < args.length; i = i + 1){
+      if (i == 1){
+        stringValue = stringValue + args[i];
+      }
+      else{
+        stringValue = stringValue + " " + args[i];
+      }
+    }
```

Comment:
args.slice(1).join(" ") should work instead of the for loop.

Reply:
Changed